### PR TITLE
Update segment replication backpressure

### LINKED ISSUE #3695: [DOC] Add Segment Replication Backpressure to documentation
**What do you want to do?**
 
- [ ] Request a change to existing documentation
- [x] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
 
OpenSearch is currently working on new segment replication feature. Segment Replication has been released as an experimental feature in opensearch 2.3. As part of Segment Replication feature we are introducing new Segment Replication Backpressure mechanism.

Implementation of this new segment replication backpressure mechanism is completed and more details can be found [here](https://github.com/opensearch-project/OpenSearch/issues/4478). This feature will be released in opensearch 2.7.

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.

### PR DESCRIPTION
**Pull request by kolchfa-aws**
Fixes #3695 

### Checklist
- [ x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _tuning-your-cluster/availability-and-recovery/segment-replication/backpressure.md [hunk 1]
```diff
@@ -7,26 +7,26 @@ has_children: false
...
+Segment replication backpressure is a shard-level rejection mechanism that dynamically rejects indexing requests as replica shards in your cluster fall behind primary shards. With segment replication backpressure, indexing requests are rejected when the percentage of stale shards in the replication group exceeds `MAX_ALLOWED_STALE_SHARDS` (50 percent by default). A replica is considered stale if it is behind the primary shard by the number of checkpoints that exceeds the `MAX_INDEXING_CHECKPOINTS` setting and its current replication lag is greater than the defined `MAX_REPLICATION_TIME_SETTING` field.
```
Comment (natebower) [inline _tuning-your-cluster/availability-and-recovery/segment-replication/backpressure.md:12]:
```suggestion
Segment replication backpressure is a shard-level rejection mechanism that dynamically rejects indexing requests as replica shards in your cluster fall behind primary shards. With segment replication backpressure, indexing requests are rejected when the percentage of stale shards in the replication group exceeds `MAX_ALLOWED_STALE_SHARDS` (50% by default). A replica is considered stale if it is behind the primary shard by the number of checkpoints that exceeds the `MAX_INDEXING_CHECKPOINTS` setting and its current replication lag is greater than the defined `MAX_REPLICATION_TIME_SETTING` field.
```

File: _tuning-your-cluster/availability-and-recovery/segment-replication/backpressure.md [hunk 2]
```diff
@@ -7,26 +7,26 @@ has_children: false
...
-MAX_INDEXING_CHECKPOINTS | Integer | The maximum number of indexing checkpoints that a replica shard can fall behind when copying from primary. Once `MAX_INDEXING_CHECKPOINTS` is breached along with `MAX_REPLICATION_TIME_SETTING`, the segment replication back-pressure mechanism is triggered. Default is `4` checkpoints.
-MAX_ALLOWED_STALE_SHARDS | Floating point | The maximum number of stale replica shards that can exist in a replication group. Once `MAX_ALLOWED_STALE_SHARDS` is breached, the segment replication back-pressure mechanism is triggered. Default is `.5`, which is 50% of a replication group.
+SEGMENT_REPLICATION_INDEXING_PRESSURE_ENABLED | Boolean | Enables the segment replication backpressure mechanism. Default is `false`.
+MAX_REPLICATION_TIME_SETTING | Time unit | The maximum time that a replica shard can take to copy from primary. Once `MAX_REPLICATION_TIME_SETTING` is breached along with `MAX_INDEXING_CHECKPOINTS`, the segment replication backpressure mechanism is initiated. Default is `5 minutes`.
```
Comment (natebower) [inline _tuning-your-cluster/availability-and-recovery/segment-replication/backpressure.md:23]:
```suggestion
MAX_REPLICATION_TIME_SETTING | Time unit | The maximum amount of time that a replica shard can take to copy from primary. Once `MAX_REPLICATION_TIME_SETTING` is breached along with `MAX_INDEXING_CHECKPOINTS`, the segment replication backpressure mechanism is initiated. Default is `5 minutes`.
```

File: _tuning-your-cluster/availability-and-recovery/segment-replication/backpressure.md [hunk 3]
```diff
@@ -7,26 +7,26 @@ has_children: false
...
+MAX_INDEXING_CHECKPOINTS | Integer | The maximum number of indexing checkpoints that a replica shard can fall behind when copying from primary. Once `MAX_INDEXING_CHECKPOINTS` is breached along with `MAX_REPLICATION_TIME_SETTING`, the segment replication backpressure mechanism is initiated. Default is `4` checkpoints.
```
Comment (natebower) [inline _tuning-your-cluster/availability-and-recovery/segment-replication/backpressure.md:24]: when copying from the primary shard?

**Review by natebower — COMMENTED** [review]
@kolchfa-aws Just a few changes. Thanks!

**Review by hdhalter — APPROVED** [review]
Looks good!

File: _tuning-your-cluster/availability-and-recovery/segment-replication/index.md [hunk 4]
```diff
@@ -64,185 +65,301 @@ curl -X PUT "$host/_cluster/settings?pretty" -H 'Content-Type: application/json'
...
+        <td>0.00%</td>
+        <td>0.00%</td>
+        <td >0.00%</td>
+        <td>0.00%</td>
+        <td>0.00%</td>
+        <td>0.00%</td>
     </tr>
+</table>
+
+As the size of the workload increases, the benefits of segment replication are amplified because the replicas are not required to index the larger dataset. In general, segment replication leads to a higher throughput at a lower resource cost than document replication in all cluster configurations, not accounting for replication lag.
```
Comment (natebower) [inline _tuning-your-cluster/availability-and-recovery/segment-replication/index.md:171]: The end of the first sentence reads as though the replicas are taking the action. Is that what we mean, or do we mean "the replicas are not required in order to index the larger dataset"? In the second sentence, we could probably just say "leads to higher throughput at lower resource cost".

**Review by natebower — APPROVED** [review]
@kolchfa-aws Approved with noted changes. Thanks!

File: _tuning-your-cluster/availability-and-recovery/segment-replication/index.md (hunk 4 above)
Comment (kolchfa-aws) [reply _tuning-your-cluster/availability-and-recovery/segment-replication/index.md:171]: Yes, in the first sentence we do mean the replicas do not have to take action.

**Pull request merged**